server/market/market.go

```diff
@@ -361,109 +472,212 @@ func (m *Market) Book() (epoch int64, buys, sells []*order.LimitOrder) {
 	m.bookMtx.Lock()
 	buys = m.book.BuyOrders()
 	sells = m.book.SellOrders()
-	epoch = m.epochIdx
+	epoch = m.bookEpochIdx
 	m.bookMtx.Unlock()
 	return
 }
 
+// PurgeBook flushes all booked orders from the in-memory book and persistent
+// storage. In terms of storage, this means changing orders with status booked
+// to status revoked.
+func (m *Market) PurgeBook() {
+	m.bookMtx.Lock()
+	defer m.bookMtx.Unlock()
+
+	// Revoke all booked orders in the DB.
+	N, err := m.storage.FlushBook(m.marketInfo.Base, m.marketInfo.Quote)
+	if err != nil {
+		log.Errorf("Failed to flush book for market %s: %v", m.marketInfo.Name, err)
+	} else {
+		log.Infof("Flushed %d booked orders for market %s", N, m.marketInfo.Name)
+		// Clear the in-memory order book to match the DB.
+		m.book.Clear()
+		// Unlock all coins. TODO: only unlock previously booked order coins, do
+		// not include coins that might belong to orders still in epoch status.
+		// This won't matter if the market is suspended, but it does does if
+		// PurgeBook is used while the market is still accepting new orders and
+		// processing epochs.
+		m.coinLockerBase.UnlockAll()
+		m.coinLockerQuote.UnlockAll()
```

Comment:
I just realized that this is not correct when there are other markets with a shared asset since these book locks are really shared asset coin lockers for just this reason.

Reply:
Addressed this in 19a9e778e98aacf59717b248def0ce38e2c2a214 by returning purged order IDs, and explicitly unlocking the coins for just those orders, not all coins for asset managed by the coin locker.